### ISSUE TITLE
FCC help page

### ISSUE THREAD

**Issue by bbsmooth** [freeCodeCamp/freeCodeCamp#49949]
### Is your feature request related to a problem? Please describe.

This request has been brewing for a while now :slightly_smiling_face: 

I'm not sure what we would call it or how we would implement it, but I strongly feel that we need a central place to document (and make available to all campers) all of the various non-obvious features that are available to assist campers in using FCC. A large percentage of these can be considered accessibility features and therefore information about them should be easily available for those who may benefit from them. Some examples (not a comprehensive list):

- The "power user" keyboard shortcuts to quickly navigate between steps
- Using `Ctrl+m` in the editor to toggle the tab trap. 
- Using `Alt + down/up arrow` to scroll the entire editor window
- Using `F1` to access the monaco built-in menu
- The workaround we put in place to deal with a bug in the JAWS screen reader when using the monaco editor
- The fact that the column of line numbers in the editor can always be scrolled by touch users

This will also allow us to clean up the interface by moving information from the UI itself to the documentation. For example:

- We can remove the `(Ctrl + Enter)` on the Check Your Code button. In fact, we already do this for narrow view ports, so this information already needs to be presented somewhere else.
- Screen reader users currently hear the same information about accessing the accessibility menu and such every time they put focus in an editor. Moving this information to a central help area will allow us to eliminate this repetitive information.

This does not necessarily need to be limited just to describing "hidden" functionality. For example, it could provide a legend for the icon buttons. It could provide a brief FAQ for the most common issues campers run into using the interface. It could provide accessibility information specific to various assistive technology.

### Describe the solution you'd like

Not sure what form this would take. Opening up for discussions. My initial thought would be to create a Help page or modal dialog and add a link to it in the main menu. 

### Describe alternatives you've considered

N/A

**Comment by naomi-lgbt** [comment]
> Using Ctrl+m in the editor to toggle the tab trap.

I needed this one myself thank you. (Not being able to tab instead of space was killing me)

**Comment by CodysCodingCloud** [comment]
I believe adding a link to the accessibility/documentation page you are describing to the help dropdown might be an intuitive place to find this information.
<img width="642" alt="image" src="https://user-images.githubusercontent.com/39781723/229955388-54d44bcd-7a63-40df-9a54-f73963ed88e8.png">

**Comment by bbsmooth** [comment]
@CodysCodingCloud but then it would only be available if you are in a step or challenge. Granted, most of the time people are in one of those, but I think it would be better if the link were universally available.

**Comment by caiweiyu** [comment]
I like this challenge very much. Please assign it to me. Thank you

**Comment by naomi-lgbt** [comment]
Going to loop @ahmadabdolsaheb in for the UX discussion.

My two cents would be to have a link in the sign-up flow, and a link available in the navigation menu.

**Comment by naomi-lgbt** [comment]
Hey @caiweiyu~! Thanks for your interest in contributing.

We typically do not assign issues. Instead, we accept the first pull request that comprehensively solves the issue.

Issues labelled with `help wanted` or `first timers only` are open for contributions. As this issue has yet to be triaged, it is not yet open for contribution.

Please make sure you read [our guidelines for contributing](https://contribute.freecodecamp.org/#/). We prioritize contributors following the instructions in our guide. Join us in [our chat room](https://discord.gg/PRyKn3Vbay) or [the forum](https://forum.freecodecamp.org/c/contributors/3) if you need help contributing - our community will be happy to assist you.

**Comment by ahmadabdolsaheb** [comment]
Could you expand on this a little bit or 
The workaround we put in place to deal with a bug in the JAWS screen reader when using the monaco editor

**Comment by ahmadabdolsaheb** [comment]
Thank you for creating this issue.

> The workaround we put in place to deal with a bug in the JAWS screen reader when using the monaco editor.

Could you link the related discussion?

To my understanding, there are two categories: shortcuts, how to use the website.
I will be working with the  mobile team to address some related issues and create an onboarding experience to make sure users know how to use the app.

We could consider a similar feature in the web app, but I want to keep the user journey as short as possible.

I think the most straight forward solution would be to create a help central page and add it to the dropdown menu.

There we could have different categories: Discover shortcuts, how to use the website, Accessibility tips, Bugs and troubleshooting, etc. There we can decide what each category does: link to updated article on news, list items, link to related forum post, etc.

**Comment by bbsmooth** [comment]
> I think the most straight forward solution would be to create a help central page and add it to the dropdown menu.

I would be down with this.

I'm pinging this issue again because I was reminded of it when testing out [MDN's new code playground](https://developer.mozilla.org/en-US/play). When evaluating new web pages, I always start by trying to use it with the keyboard only. Besides their annoying main menu that automatically opens the submenus as you tab through it, the major annoyance was that once I tabbed into the HTML editor I couldn't find any way to get out of it. I had to spend quite some time until I figured out a solution. I don't want to tell you what it is because I don't want to spoil the fun just in case you want to try.

The point being that the solution was not obvious to me (I basically stumbled upon it by luck), and even more important, I could not find any documentation/help on the page. We can do better than MDN :slightly_smiling_face:

**Comment by DylanDevelops** [comment]
I agree with having documentation related to how to make your FCC accessible on the platform you are using. Let me know if there is anything I can do to help!

**Comment by ahmaxed** [comment]
Sounds like plan. Here is my proposed implementation:
- have a help item in the dropdown menu
- the help page would have a grid of items: accessibility tips, how to use the website, and troubleshooting.
- each item in the grid will have a title and and short description.
- each of the grid items would link to a page that would include the related information.
If that makes sense, I could do a mock and we could open this issue to contributors.

**Comment by ahmaxed** [comment]
I don't have a preference, but having them on learn might help with localization.

**Comment by VaibhavSharma24** [comment]
@ahmaxed i think the idea you proposed of adding the help page in dropdown menu which consists accessibility tips, how to use the website, and troubleshooting with title and short description can really save time and help users to find out the shortcuts and help them troubleshoot if they are stuck I think this page can really enhance the user experience. 

Can I put a preview of how it will look ?

**Comment by ahmaxed** [comment]
@VaibhavSharma24, feel free to put together an initial wire-frame or a mock.

**Comment by VaibhavSharma24** [comment]
We can design it more better and can add bullets points and contact us link if problem didn't get solved through the documentation and can hide bulleted points in read more section if there is much content is it okay ? @ahmaxed 
![FreeCodeCamp__Preview](https://github.com/freeCodeCamp/freeCodeCamp/assets/72144747/444af445-b3a7-4806-a472-91d0a1fbf69c)

**Comment by ahmaxed** [comment]
Yes, basically the help page would be a list of three pages (troubleshooting, how to use the website, Accessibility tips)

**Comment by ahmaxed** [comment]
Here is a quick mock:
<img width="768" alt="Screenshot 2023-10-20 at 12 49 44 PM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/4591597/b3469ef9-7817-4683-bde9-07412a20d81b">

The icons, titles, and descriptions could be modified to describe each item better.

I suggest deciding what will go on each page first and then creating them individually.

Once that is done, we could create the help page to link to them.

The layout for each page could be similar to superblock pages.

**Comment by bbsmooth** [comment]
I'm so happy to see this taking shape. How are we going to collaborate on creating these pages? As you can imagine, I have a lot of topics for each one.

**Comment by ahmaxed** [comment]
List the page titles and their descriptions. Then, describe what you want each page to include . That way we could look at the data and decide on the layout of each page. Does that sound good?

**Comment by naomi-lgbt** [comment]
I think we can go ahead and start building this out with the mock Ahmad provided above.

I'll throw together some quick articles next week for those to link to, for now.

**Comment by Sembauke** [comment]
@naomi-lgbt ping

**Comment by a2937** [comment]
Would the content linked in the news article be a good starting point for the FAQ page? https://www.freecodecamp.org/news/about/

**Comment by ahmaxed** [comment]
yes, that should work. 
We can add the other items and create related articles/pages for them.

**Comment by a2937** [comment]
Oh poop. We already have Frequently added questions on the landing page. I forgot.

**Comment by ahmaxed** [comment]
Yes, but that list is limited and at the bottom of the landing page.

**Comment by ahmaxed** [comment]
@a2937, the faq section of the landing page would not be sufficient to serve as a FAQ page for the site as it was designed for landing page conversion optimization.

I think we should wait for Noami to work on dedicated articles on for the help page.

**Comment by naomi-lgbt** [comment]
This is on my list to get done this week.

**Comment by X1Vi** [comment]
@naomi-lgbt hey I notice you are working on this from Nov 13, 2024. Are you still working on this presently ?

**Comment by naomi-lgbt** [comment]
Yes.

**Comment by hbar1st** [comment]
Just want to mention, if it hasn't already been said, that adding a section for the Exams would help users who have some common questions about them. (similar to what is in this article: https://www.freecodecamp.org/news/freecodecamps-new-javascript-certification-is-now-live/

But also, having so many articles that cover different aspects of fCC is getting hard to manage/remember even for a mod. For eg. here's another article <eg. <https://www.freecodecamp.org/news/support/> that talks about technical support things like account issues, and yet another article that is more generally about how fCC works <https://www.freecodecamp.org/news/about> and yet another article introducing the discord server <https://www.freecodecamp.org/news/freecodecamp-discord-chat-room-server/> and yet one more about contributing to fCC <https://contribute.freecodecamp.org/intro/> and another about how to claim your supporter role on discord <https://www.freecodecamp.org/news/discord-supporter-role/> and, ... yeah, do I need to go on? XD
